Fixing persistence of passwords

Doom4535 opened the pull request:
Adding fix suggested by: @solvisit in issue: https://github.com/SenseTecnic/node-red-contrib-users/issues/1

This closes SenseTecnic/node-red-contrib-users/issues/1

Doom4535 committed:
Fixing persistence of passwords

Adding fix suggested by: @solvisit in issue: https://github.com/SenseTecnic/node-red-contrib-users/issues/1

This closes SenseTecnic/node-red-contrib-users/issues/1

Doom4535 committed:
Starting flow based credential editor

Doom4535 committed:
Hunting for bug preventing node display in ui/dash

Doom4535 committed:
Fixing naming bug

Doom4535 committed:
Starting basic node
Following setup guide from first-node guide

Doom4535 committed:
Starting on making custom node
Need to add in error handling for messages (see section at the
beginning, remember to use locals for language support)
Need to add in logic, identify functions to use

Doom4535 committed:
Connecting the User Credential Manager Node
Connected the User Credential Manager Node to access the user account
objects.  Do note that the passwords are stored as a hash, so they are
not able to be displayed; however, this gives us the access we need to
enable overwriting of the passwords and the creation of new accounts.

Doom4535 committed:
Defining procedure for handling messages
The locales/en-US/users_manager.json may not be included properly

Doom4535 committed:
Squashed commit of the following:

commit 1d3ef52302fd5603b235ac8c8898e28efb9ca231
Author: Linux User <aarcov@gmail.com>
Date:   Tue Apr 23 21:10:26 2019 +0000

    Defining procedure for handling messages
    The locales/en-US/users_manager.json may not be included properly

commit 601dc5c12d086bc1f30582fb87c5f897efd0d378
Author: Linux User <aarcov@gmail.com>
Date:   Tue Apr 23 19:51:18 2019 +0000

    Connecting the User Credential Manager Node
    Connected the User Credential Manager Node to access the user account
    objects.  Do note that the passwords are stored as a hash, so they are
    not able to be displayed; however, this gives us the access we need to
    enable overwriting of the passwords and the creation of new accounts.

commit ee377f62ad0f83ffeb25bf925dd52f1910d424e5
Author: Linux User <aarcov@gmail.com>
Date:   Tue Apr 23 17:06:15 2019 +0000

    Starting on making custom node
    Need to add in error handling for messages (see section at the
    beginning, remember to use locals for language support)
    Need to add in logic, identify functions to use

Doom4535 committed:
Modifying to expose get and add user functions
Exposed existing getUser()
Exposed hash()

Added addUser()
Added getUserExistance()
Added getUserAccount()

Also added the following, one of which breaks something:
Added updateUser()
Added deleteUser()

Doom4535 committed:
Squashed commit of the following:

commit 4926852925ce18eca1278fd0dd9aecf714d2971a
Author: Linux User <aarcov@gmail.com>
Date:   Wed Apr 24 15:52:28 2019 +0000

    Modifying to expose get and add user functions
    Exposed existing getUser()
    Exposed hash()

    Added addUser()
    Added getUserExistance()
    Added getUserAccount()

    Also added the following, one of which breaks something:
    Added updateUser()
    Added deleteUser()

commit 1d3ef52302fd5603b235ac8c8898e28efb9ca231
Author: Linux User <aarcov@gmail.com>
Date:   Tue Apr 23 21:10:26 2019 +0000

    Defining procedure for handling messages
    The locales/en-US/users_manager.json may not be included properly

commit 601dc5c12d086bc1f30582fb87c5f897efd0d378
Author: Linux User <aarcov@gmail.com>
Date:   Tue Apr 23 19:51:18 2019 +0000

    Connecting the User Credential Manager Node
    Connected the User Credential Manager Node to access the user account
    objects.  Do note that the passwords are stored as a hash, so they are
    not able to be displayed; however, this gives us the access we need to
    enable overwriting of the passwords and the creation of new accounts.

commit ee377f62ad0f83ffeb25bf925dd52f1910d424e5
Author: Linux User <aarcov@gmail.com>
Date:   Tue Apr 23 17:06:15 2019 +0000

    Starting on making custom node
    Need to add in error handling for messages (see section at the
    beginning, remember to use locals for language support)
    Need to add in logic, identify functions to use

Doom4535 committed:
Merge branch 'local-dev' into 5-creating-user-from-flow
Merging in development changes

Doom4535 committed:
Squashing bugs, adding in methods
Fixed issue with checking for user existance
Fixed issue with createUser not properly checking for if a user already exists
 Added in method to delete users

Doom4535 committed:
Basic flow user management, closes #5
Documentation needs to be updated
Update/uncomment checkedRequiredFields in nodes/users_manager.js
Does not signout users if their credentials have been deleted (only
prevents reauthentication)

Doom4535 committed:
Merge pull request #1 from Doom4535/5-creating-user-from-flow

5 creating user from flow

tedhuang commented:
Thank you for the PR, sorry for the extreme delay in this reply. I see a lot of changes in PR that don't seem to be relevant to the fix itself. Is it possible to reduce this PR to the core of the fix?

Doom4535 committed:
Update package.json

This is in response to a GutHub Security advisory involving bcrypt (https://github.com/advisories/GHSA-5wg4-74h6-q47v) and has not been tested.

Doom4535 commented:
Welp, this has been a while, looks like the issue was already solved so I'll go ahead and close this
